lib/matplotlib/patches.py

```diff
@@ -3218,23 +3218,108 @@ class _Curve(_Base):
+        beginarrow = endarrow = 'none'  # Whether arrows are drawn.
         fillbegin = fillend = False  # Whether arrows are filled.
 
-        def __init__(self, head_length=.4, head_width=.2):
+        def __init__(self, head_length=.4, head_width=.2, widthA=1., widthB=1.,
+                     lengthA=0.2, lengthB=0.2, angleA=0, angleB=0, scaleA=None,
+                     scaleB=None):
             """
             Parameters
             ----------
             head_length : float, default: 0.4
                 Length of the arrow head, relative to *mutation_scale*.
             head_width : float, default: 0.2
                 Width of the arrow head, relative to *mutation_scale*.
+            widthA : float, default: 1.0
+                Width of the bracket at the beginning of the arrow
+            widthB : float, default: 1.0
+                Width of the bracket at the end of the arrow
+            lengthA : float, default: 0.2
+                Length of the bracket at the beginning of the arrow
+            lengthB : float, default: 0.2
+                Length of the bracket at the end of the arrow
+            angleA : float, default 0
+                Orientation of the bracket at the beginning, as a
+                counterclockwise angle. 0 degrees means perpendicular
+                to the line.
+            angleB : float, default 0
+                Orientation of the bracket at the beginning, as a
+                counterclockwise angle. 0 degrees means perpendicular
+                to the line.
+            scaleA : float, default *mutation_size*
+                The mutation_size for the beginning bracket
+            scaleB : float, default *mutation_size*
+                The mutation_size for the end bracket
             """
+
             self.head_length, self.head_width = head_length, head_width
+            self.widthA, self.widthB = widthA, widthB
+            self.lengthA, self.lengthB = lengthA, lengthB
+            self.angleA, self.angleB = angleA, angleB
+            self.scaleA, self.scaleB = scaleA, scaleB
+
+            if self.beginarrow == 'arrow':
+                self._beginarrow_head = True
+                self._beginarrow_bracket = False
+            elif self.beginarrow == 'bracket':
+                self._beginarrow_head = False
+                self._beginarrow_bracket = True
+            elif self.beginarrow == 'none':
+                self._beginarrow_head = False
+                self._beginarrow_bracket = False
+            elif self.beginarrow:
+                self._beginarrow_head = True
+                self._beginarrow_bracket = False
+
+                _api.warn_deprecated('3.5', message="The use of boolean values"
+                                     "in beginarrow is now decapitated, prefer"
+                                     "using the string 'arrow' for the same"
+                                     "function")
+            elif not self.beginarrow:
```

Comment:
You really need to check the type here, and not just truthiness. Any string is truthy, except '', which will not trigger the final error as it should.